ui/packages/app/web/src/components/ui/ThemeToggle.tsx

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {useCallback, useEffect, useState} from 'react';
+
+import {Icon} from '@iconify/react';
+
+import {Dropdown, IconButton} from '@parca/components';
+import {
+  selectDarkMode,
+  selectParcaThemeSystemSettings,
+  setDarkMode,
+  setParcaThemeSystemSettings,
+  useAppDispatch,
+  useAppSelector,
+} from '@parca/store';
+
+const ThemeToggle = () => {
+  const dispatch = useAppDispatch();
+  const isDarkMode = useAppSelector(selectDarkMode);
+  const isSystemSettingsTheme = useAppSelector(selectParcaThemeSystemSettings);
+  const [systemSettingsDarkMode, setSystemSettingsDarkMode] = useState(false);
+
+  useEffect(() => {
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+
+    // Listen for changes to the prefers-color-scheme media query and then update the theme accordingly.
+    mediaQuery.onchange = media => {
+      if (mediaQuery.matches === true) {
+        setSystemSettingsDarkMode(true);
+      } else {
+        setSystemSettingsDarkMode(false);
+      }
+    };
+  });
```

Comment:
I suppose the dependency array `[]` is missed here? Or thats intentional?

Reply:
added that bit 👍🏾